hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/StreamJmsP.java

```diff
@@ -230,8 +230,12 @@ protected void restoreFromSnapshot(@Nonnull Object key, @Nonnull Object value) {
 
     @Override
     public void close() throws Exception {
-        consumer.close();
-        session.close();
+        if (consumer != null) {
+            consumer.close();
```

Comment:
can this throw?

Reply:
yes, "due to some internal error", as the javadoc says. If it throws, we won't throw the session, but we'll throw the connection in the PSupplier, which should close everything...